File: cli-tools/platform/node/add-color-library.js
const axios = require('axios');
const { BASE_URL, getAuthParams } = require('./utils');

const imageUrls = process.argv.slice(2);

if (imageUrls.length === 0) {
    console.log("Usage: node add-color-library.js <url> [url] ...");
    process.exit(1);
}

async function addColorLibrary() {
    try {
        const auth = getAuthParams();
        const payload = {
            ...auth,
            imagesUrl: imageUrls
        };

        const response = await axios.post(`${BASE_URL}/AddColorLibrary`, payload);
        console.log(JSON.stringify(response.data, null, 2));

    } catch (error) {
        console.error("Error adding color library:", error.message);
        if (error.response) console.error(error.response.data);
    }
}

addColorLibrary();

File: cli-tools/platform/python/add_color_library.py
import argparse
import requests
import json
import sys
from utils import BASE_URL, get_auth_params

def add_color_library(image_urls):
    try:
        auth = get_auth_params()
        payload = auth.copy()
        payload['imagesUrl'] = image_urls

        response = requests.post(f"{BASE_URL}/AddColorLibrary", json=payload)
        response.raise_for_status()

        print(json.dumps(response.json(), indent=2))

    except Exception as e:
        print(f"Error adding color library: {e}", file=sys.stderr)

if __name__ == "__main__":
    parser = argparse.ArgumentParser(description='Register color swatches with Pixelz Platform API')
    parser.add_argument('urls', nargs='+', help='One or more swatch image URLs')
    args = parser.parse_args()

    add_color_library(args.urls)
